### PR TITLE
Update Install and Examples Files for Streams 2.5

### PR DESCRIPTION
**Pull request by OwenCorrigan76**
Update install Files for 2.5.0 release
@kyguy

### CODE REVIEW

File: install/DO_NOT_EDIT.md [hunk 1]
```diff
@@ -0,0 +1 @@
+These install files are for the last stable Strimzi release. They should be updated only when a new Strimzi version is released, not as a part of regular PR. The editable files for the next release can be found in the [`packaging`](../packaging/) directory.
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply install/DO_NOT_EDIT.md]: Removed

File: install/canary/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Strimzi Canary
+
+Strimzi canary is a tool which acts as an indicator of whether Kafka clusters are operating correctly.
+This is achieved by creating a canary topic and periodically producing and consuming events on the topic and getting metrics out of these exchanges.
+
+For more information and installation guide, see [https://github.com/strimzi/strimzi-canary](https://github.com/strimzi/strimzi-canary).
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply install/canary/README.md]: Removed

File: install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml [hunk 3]
```diff
@@ -55,55 +55,45 @@ spec:
...
-              value: registry.redhat.io/amq-streams/strimzi-rhel8-operator:2.4.0
+              value: registry.redhat.io/amq-streams/strimzi-rhel8-operator:2.5.0
             - name: STRIMZI_DEFAULT_KAFKA_INIT_IMAGE
-              value: registry.redhat.io/amq-streams/strimzi-rhel8-operator:2.4.0
+              value: registry.redhat.io/amq-streams/strimzi-rhel8-operator:2.5.0
             - name: STRIMZI_DEFAULT_KAFKA_BRIDGE_IMAGE
-              value: registry.redhat.io/amq-streams/bridge-rhel8:2.4.0
+              value: registry.redhat.io/amq-streams/bridge-rhel8:2.5.0
             - name: STRIMZI_DEFAULT_MAVEN_BUILDER
               value: registry.access.redhat.com/ubi8/openjdk-11:1.10
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml]: Is this correct?

File: install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml [hunk 4]
```diff
@@ -117,92 +107,103 @@ spec:
...
+              value: ""
+            - name: STRIMZI_LEADER_ELECTION_ENABLED
+              value: "true"
+            - name: STRIMZI_LEADER_ELECTION_LEASE_NAME
+              value: "strimzi-cluster-operator"
+            - name: STRIMZI_LEADER_ELECTION_LEASE_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: STRIMZI_LEADER_ELECTION_IDENTITY
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml]: Done

File: install/drain-cleaner/certmanager/070-ValidatingWebhookConfiguration.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      service:
+        namespace: "strimzi-drain-cleaner"
+        name: "strimzi-drain-cleaner"
+        path: /drainer
+        port: 443
+      caBundle: Cg==
+    admissionReviewVersions: ["v1"]
+    sideEffects: None
+    failurePolicy: Ignore
+    timeoutSeconds: 5
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply install/drain-cleaner/certmanager/070-ValidatingWebhookConfiguration.yaml]: Deleted

File: install/drain-cleaner/kubernetes/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+4) With the installation files prepared, deploy Strimzi Drain Cleaner:
+   
+   ```
+   kubectl apply -f ./
+   ```
+
+## Certificate renewals
+
+Users are responsible for renewing the certificates before they expire.
+The certificates can be renewed by just repeating the whole process and applying the updated YAML files.
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply install/drain-cleaner/kubernetes/README.md]: Deleted

File: install/drain-cleaner/openshift/060-Deployment.yaml [hunk 7]
```diff
@@ -4,6 +4,13 @@ metadata:
   name: strimzi-drain-cleaner
   labels:
     app: strimzi-drain-cleaner
+    com.company: Red_Hat
+    rht.prod_name: Red_Hat_Application_Foundations
+    rht.prod_ver: "2023.Q3"
+    rht.comp: AMQ_Streams
+    rht.comp_ver: "2.5"
+    rht.subcomp: drain-cleaner
+    rht.subcomp_t: application
```
[the remark this answers is not in the document]
Comment (OwenCorrigan76) [reply install/drain-cleaner/openshift/060-Deployment.yaml]: Done

**Comment by OwenCorrigan76** [comment]
@kyguy - Addressed first round of comments. Re requesting review.

File: install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml [hunk 8]
```diff
@@ -99,11 +99,12 @@ spec:
...
-            - name: STRIMZI_LEADER_ELECTION_IDENTITY
-              valueFrom:
-                fieldRef:
-                  fieldPath: metadata.name
+                   fieldPath: metadata.namespace
+           - name: STRIMZI_LEADER_ELECTION_IDENTITY
+              spec:
+               valueFrom:
+                 fieldRef:
+                   fieldPath: metadata.name
```
Comment (ppatierno) [inline install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml:107]: I think the env var declaration `STRIMZI_LEADER_ELECTION_IDENTITY` is not aligned correctly.

File: install/drain-cleaner/openshift/060-Deployment.yaml [hunk 9]
```diff
@@ -15,17 +15,17 @@ spec:
...
-        rht.comp: AMQ_Streams
-        rht.comp_ver: "2.4"
-        rht.subcomp: drain-cleaner
-        rht.subcomp_t: application
+          rht.prod_name: Red_Hat_Application_Foundations
+          rht.prod_ver: "2023.Q3"
+          rht.comp: AMQ_Streams
+          rht.comp_ver: "2.5"
+          rht.subcomp: drain-cleaner
+          rht.subcomp_t: application
```
Comment (ppatierno) [inline install/drain-cleaner/openshift/060-Deployment.yaml:23]: wrong indentation?

**Review by kyguy — APPROVED** [review]
This looks great, just need to fix the comments from Paolo and we can merge, then open a follow up PR for the example files

File: install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml (hunk 8 above)
Comment (OwenCorrigan76) [reply install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml]: Fixed

File: install/drain-cleaner/openshift/060-Deployment.yaml (hunk 9 above)
Comment (OwenCorrigan76) [reply install/drain-cleaner/openshift/060-Deployment.yaml]: @ppatierno I was in the wrong yaml. Fixed now!!

File: install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml [hunk 10]
```diff
@@ -99,7 +99,7 @@ spec:
             - name: STRIMZI_LEADER_ELECTION_LEASE_NAMESPACE
               valueFrom:
                 fieldRef:
-                  fieldPath: metadata.namespace
+                   fieldPath: metadata.namespace
```
Comment (kyguy) [inline install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml:102]: Remove added space

Comment (OwenCorrigan76) [reply install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml]: Done

**Comment by OwenCorrigan76** [comment]
Backported Examples directory added

File: examples/DO_NOT_EDIT.md [hunk 11]
```diff
@@ -0,0 +1 @@
+These example files are for the last stable Strimzi release. They should be updated only when a new Strimzi version is released, not as a part of regular PR. The editable files for the next release can be found in the [`packaging`](../packaging/) directory.
```
Comment (ppatierno) [inline examples/DO_NOT_EDIT.md:1]: we are not bringing this file because it has details about `packaging` folder which is related to Strimzi. You should remove it.

File: examples/README.md [hunk 12]
```diff
@@ -16,5 +16,6 @@ This folder contains different examples of AMQ Streams custom resources and demo
     * Prometheus Alert
     * Sample Prometheus installation files
     * Sample Grafana installation files
+    * JMX Trans deployment
```
Comment (ppatierno) [inline examples/README.md:19]: JMX Trans is not something we have in AMQ Streams, you should remove it.

File: examples/README.md [hunk 13]
```diff
@@ -16,5 +16,6 @@ This folder contains different examples of AMQ Streams custom resources and demo
...
 * [Security](./security)
-    * Deployments of Kafka, Kafka Connect and HTTP Bridge using TLS encryption, authentication and authorization
+    * Deployments of Kafka, Kafka Connect and HTTP Bridge using TLS encryption, authentication and authorization
```
Comment (ppatierno) [inline examples/README.md:21]: not clear what's the difference here :-)

File: examples/connect/kafka-connect-build.yaml [hunk 14]
```diff
@@ -0,0 +1,43 @@
+apiVersion: kafka.strimzi.io/v1beta2
```
Comment (ppatierno) [inline examples/connect/kafka-connect-build.yaml:1]: also this file it should not be here because it's leveraging a temporary registry (24h) where we push Kafka Connect builds. You should remove it, can you confirm @kyguy ? I remember we had this discussion with Jakub in the past as well.

File: examples/security/keycloak-authorization/README.md [hunk 15]
```diff
@@ -1,17 +1,22 @@
...
 This folder contains an example `Kafka` custom resource configured for OAuth 2.0 token-based authorization using Keycloak. The example resource is configured with:
+
 - `keycloak` authorization
 - the corresponding `oauth` authentication
+
 The folder also contains a Keycloak realm export to import into your Keycloak instance to support the example.
 
-Full instructions for Trying Red Hat Single Sign-On Authorization Services are described in the [documentation for deploying and managing AMQ Streams on OpenShift](https://access.redhat.com/documentation/en-us/red_hat_amq_streams/).
+Full instructions for Trying Red Hat Single Sign-On Authorization Services are described in the [documentation for deploying and managing AMQ Streams on OpenShi
+ft](https://access.redhat.com/documentation/en-us/red_hat_amq_streams/).
```
Comment (ppatierno) [inline examples/security/keycloak-authorization/README.md:11]: why this change?

File: examples/security/keycloak-authorization/README.md [hunk 16]
```diff
@@ -1,17 +1,22 @@
 # Keycloak authorization example
 
 This folder contains an example `Kafka` custom resource configured for OAuth 2.0 token-based authorization using Keycloak. The example resource is configured with:
+
 - `keycloak` authorization
 - the corresponding `oauth` authentication
+
```
Comment (ppatierno) [inline examples/security/keycloak-authorization/README.md:7]: why this blank line?

File: examples/security/keycloak-authorization/README.md [hunk 17]
```diff
@@ -1,17 +1,22 @@
 # Keycloak authorization example
 
 This folder contains an example `Kafka` custom resource configured for OAuth 2.0 token-based authorization using Keycloak. The example resource is configured with:
+
```
Comment (ppatierno) [inline examples/security/keycloak-authorization/README.md:4]: why this blank line?

File: examples/security/keycloak-authorization/README.md [hunk 18]
```diff
@@ -1,17 +1,22 @@
...
-Full instructions for Trying Red Hat Single Sign-On Authorization Services are described in the [documentation for deploying and managing AMQ Streams on OpenShift](https://access.redhat.com/documentation/en-us/red_hat_amq_streams/).
+Full instructions for Trying Red Hat Single Sign-On Authorization Services are described in the [documentation for deploying and managing AMQ Streams on OpenShi
+ft](https://access.redhat.com/documentation/en-us/red_hat_amq_streams/).
 
 - [kafka-authz-realm.json](./kafka-authz-realm.json)
   - The Keycloak realm export file
 - [kafka-ephemeral-oauth-single-keycloak-authz.yaml](./kafka-ephemeral-oauth-single-keycloak-authz.yaml)
   - The Kafka CR that defines a single-node Kafka cluster with `oauth` authentication and `keycloak` authorization,
-    using the `kafka-authz` realm.
+    using the `kafka-authz` realm. See [full example instructions](https://strimzi.io/docs/operators/0.36.0/configuring.html#proc-oauth-authorization-keycloak-example_str) for proper preparation and deployment.
```
Comment (ppatierno) [inline examples/security/keycloak-authorization/README.md:17]: not sure we should have a link to upstream documentation here. @kyguy @scholzj @PaulRMellor ?

File: examples/security/keycloak-authorization/README.md [hunk 19]
```diff
@@ -1,17 +1,22 @@
...
 - [kafka-ephemeral-oauth-single-keycloak-authz-metrics.yaml](./kafka-ephemeral-oauth-single-keycloak-authz-metrics.yaml)
   - The Kafka CR that defines a single-node Kafka cluster with `oauth` authentication and `keycloak` authorization,
     with included configuration for exporting the OAuth metrics using Prometheus JMX exporter.
+
+More examples are available at the oauth module [example documentation](https://github.com/strimzi/strimzi-kafka-oauth/tree/main/examples).
```
Comment (ppatierno) [inline examples/security/keycloak-authorization/README.md:22]: Ditto as above.

File: examples/connect/kafka-connect-build.yaml (hunk 14 above)
Comment (kyguy) [reply examples/connect/kafka-connect-build.yaml:1]: @ppatierno you are right [1]

[1] https://github.com/jboss-container-images/amqstreams-1-openshift-image/pull/382

File: examples/DO_NOT_EDIT.md (hunk 11 above)
Comment (OwenCorrigan76) [reply examples/DO_NOT_EDIT.md]: Removed

File: examples/README.md (hunk 12 above)
Comment (OwenCorrigan76) [reply examples/README.md]: Removed

File: examples/README.md (hunk 13 above)
Comment (OwenCorrigan76) [reply examples/README.md:20]: Not sure why this was in the diff??

File: examples/connect/kafka-connect-build.yaml (hunk 14 above)
Comment (OwenCorrigan76) [reply examples/connect/kafka-connect-build.yaml:1]: Removed

File: examples/security/keycloak-authorization/README.md (hunk 17 above)
Comment (OwenCorrigan76) [reply examples/security/keycloak-authorization/README.md]: Removed

**Comment by kyguy** [comment]
Merged as part of this [1]

[1] https://github.com/jboss-container-images/amqstreams-1-openshift-image/pull/414
